pkg/util/github/repo.go

```diff
@@ -140,5 +140,39 @@ func (c *Client) PushInitRepo(transitBranch, branch, localPath string) error {
 
 	// 2. push local path to repo
 	needRollBack, err := c.PushLocalPathToBranch(transitBranch, branch, localPath)
+	if err != nil {
+		return err
+	}
+
+	// 3. protect branch
+	err = c.ProtectBranch(branch)
 	return err
 }
+
+// ProtectBranch will protect the special branch
+func (c *Client) ProtectBranch(branch string) error {
+	req := &github.ProtectionRequest{
+		EnforceAdmins: false,
+		RequiredPullRequestReviews: &github.PullRequestReviewsEnforcementRequest{
+			RequireCodeOwnerReviews:      true,
+			DismissStaleReviews:          true,
+			RequiredApprovingReviewCount: 1,
+		},
+		RequiredConversationResolution: github.Bool(true),
+	}
+
+	repo, err := c.GetRepoDescription()
+	if err != nil {
+		log.Errorf("Get repo failed: %s.", err)
```

Comment:
If this function returns an error to upstream, do not print the error, as this will cause the error to be printed twice.

All similarities need to be modified. Thank you for your hard work!

Reply:
My bad, Golang is not my most used language, thanks for your suggestion

Reply:
line 171 need to be modified as well.

Reply:
That's nothing, it's our greatest joy to see community contributions.